public/v4/apps/shiori.yml

```diff
@@ -1,40 +1,73 @@
 captainVersion: 4
-
 services:
+    # Shiori
     $$cap_appname:
-        image: radhifadlillah/shiori@sha256:3d1eb1f0aeb9804dea41f3ec5d29de5cf214bb66ef04dce1aa942ca17caec7bb
+        image: ghcr.io/go-shiori/shiori:$$cap_shiori_version
         environment:
-            SHIORI_DIR: /data
+            SHIORI_DBMS: mysql
+            SHIORI_MYSQL_USER: $$cap_mariadb-user
+            SHIORI_MYSQL_PASS: $$cap_mariadb-pass
+            SHIORI_MYSQL_NAME: $$cap_mariadb-db
+            SHIORI_MYSQL_ADDRESS: tcp(srv-captain--$$cap_appname-mariadb)
         volumes:
-            - $$cap_appname-shiori-data:/data
-        restart: on-failure
+            - $$cap_appname-data:/shiori
+        restart: unless-stopped
         caproverExtra:
             containerHttpPort: '8080'
-
+    # MariaDB
+    $$cap_appname-mariadb:
+        image: mariadb:$$cap_mariadb_version
+        environment:
+            MYSQL_RANDOM_ROOT_PASSWORD: '1'
+            MYSQL_DATABASE: $$cap_mariadb-db
+            MYSQL_USER: $$cap_mariadb-user
+            MYSQL_PASSWORD: $$cap_mariadb-pass
+        volumes:
+            - $$cap_appname-mariadb-data:/var/lib/mysql
+        restart: unless-stopped
+        caproverExtra:
+            notExposeAsWebApp: 'true'
 caproverOneClickApp:
+    variables:
+        - id: $$cap_shiori_version
+          label: Shiori Version Tag
+          description: Check out their Docker page for the valid tags https://github.com/go-shiori/shiori/pkgs/container/shiori/versions?filters%5Bversion_type%5D=tagged
+          defaultValue: 'v1.5.5'
+        - id: $$cap_mariadb_version
+          label: MariaDB Version
+          defaultValue: '11.0.2'
+          description: Check out their docker page for the valid tags https://hub.docker.com/r/library/mariadb/tags/
+          validRegex: /^([^\s^\/])+$/
+        - id: $$cap_mariadb-db
+          label: MariaDB Database
+          description: Database name for Shiori
+          defaultValue: 'shiori'
+        - id: $$cap_mariadb-user
+          label: MariaDB User
+          description: Database User for Shiori
+          defaultValue: 'shiori'
+        - id: $$cap_mariadb-pass
```

Comment:
Suggest using `$$cap_gen_random_hex(16)` as a default value for the ease of installation